src/app/content/utils/seoUtils.ts

```diff
@@ -25,6 +27,54 @@ const getParentPrefix = (node: LinkedArchiveTreeNode | undefined): string => {
 
 };
 
+const hideMath = (node: any) => {
+  const mathSpans = node.querySelectorAll('.os-math-in-para');
+  mathSpans.forEach((el: any) => {
+    el.outerHTML = '...';
+  });
+  return node;
+};
+
+const getPageType = (node: any) => {
+  if (!node) {
+    return '';
+  }
+
+  if (node.classList.contains('appendix')) {
+    return 'appendix';
+  } else {
+    return node.getAttribute('data-type');
+  }
+};
+
+export const createDescription = (pageContent: string, book: Book, page: Page) => {
+  const doc = domParser.parseFromString(pageContent, 'text/html');
+  const contentNode = doc.body.children[0];
+  const pageType = getPageType(contentNode);
+  const node = assertDefined(
+    findArchiveTreeNodeById(book.tree, page.id),
+    `couldn't find node for a page id: ${page.id}`
+  );
+  const sectionTitle = getArchiveTreeSectionTitle(node);
+  const nodeTitleDoc = domParser.parseFromString(node.title, 'text/html');
+  const titleNode = nodeTitleDoc.body.children[0];
+  const chapterFromTitle = titleNode.innerText.split('.')[0] || '';
+  const isAnswerKey = contentNode.classList.contains('os-solution-container')
+    || contentNode.classList.contains('os-solutions-container');
```

Comment:
@TomWoodward These are the two classes I've seen applied to answer keys.

Reply:
lol the pluralization. ok.